sympy/core/power.py

```diff
@@ -71,9 +71,12 @@ def integer_nthroot(y, n):
         raise ValueError("y must be nonnegative")
     if n < 1:
         raise ValueError("n must be positive")
-    if HAS_GMPY:
-        x, t = gmpy.iroot(y, n)
-        return as_int(x), t
+    if HAS_GMPY and n < 2**63:
```

Comment:
Can we directly write the result of `2**63` and mention in the comments that the result is 2**63?

Reply:
For `n < 2**63` ,`gmpy` produces `TypeError`, So for that I have added a check for `n`

Reply:
I think both CPython and pypy will constant-fold the `2**63` so that it isn't actually computed at runtime:
```julia
In [1]: def f(n): 
   ...:     if n < 2**63: 
   ...:         return 1 
   ...:     else: 
   ...:         return 0 
   ...:                                                                                                                                        

In [2]: import dis                                                                                                                             

In [3]: dis.dis(f)                                                                                                                             
  2           0 LOAD_FAST                0 (n)
              2 LOAD_CONST               1 (9223372036854775808)
              4 COMPARE_OP               0 (<)
              6 POP_JUMP_IF_FALSE       12

  3           8 LOAD_CONST               2 (1)
             10 RETURN_VALUE

  5     >>   12 LOAD_CONST               3 (0)
             14 RETURN_VALUE
             16 LOAD_CONST               0 (None)
             18 RETURN_VALUE
```

Reply:
But using it is passing all the tests locally too.

Reply:
So should I replace the `2**63` by `9223372036854775808` ? Would it work then?